lib/statsd/instrument/udp_sink.rb

```diff
@@ -8,55 +11,74 @@ def self.for_addr(addr)
     new(host, Integer(port_as_string))
   end
 
+  class Dispatcher
+    def initialize(host, port, queue)
+      @host = host
+      @port = port
+      @queue = queue
+      @socket = nil
+      @thread = nil
+      ObjectSpace.define_finalizer(self, Proc.new { |o| o.stop })
```

Comment:
Maybe use `:stop.to_proc` to ensure you're not enclosing anything in the proc scope? I don't think it's a problem right now, but at least `to_proc` would make it more clear.